src/os/bluestore/BlockDevice.h

```diff
@@ -102,8 +108,8 @@ class BlockDevice {
 
   virtual void aio_submit(IOContext *ioc) = 0;
 
-  virtual uint64_t get_size() const = 0;
-  virtual uint64_t get_block_size() const = 0;
+  virtual uint64_t get_size() const { return size; }
+  virtual uint64_t get_block_size() const { return block_size; }
```

Comment:
why the need for `virtual` here ? They are public and public inherited in the derived classes.